Updates docs for smart contracts (#8)
aragon/builders-portal: docs/smart-contracts/core/plugin/PluginCloneable.md

## Aragon Core

###  contract `PluginCloneable`

An abstract, non-upgradeable contract to inherit from when creating a plugin
being deployed via the minimal clones pattern (see [ERC-1167](https://eips.ethereum.org/EIPS/eip-1167)).

#### public variable `PLUGIN_INTERFACE_ID`

```solidity
bytes4 PLUGIN_INTERFACE_ID 
```

####  error `ClonesInitAlreadyInitialized`

```solidity
error ClonesInitAlreadyInitialized() 
```

#### external function `__PluginCloneable_init`

used by the aragon to call by default for PluginCloneable interfaces.

```solidity
function __PluginCloneable_init(address _dao) external 
```

| Input | Type | Description |
|:----- | ---- | ----------- |
| _dao | address | the dao address to set in a slot. |

#### internal modifier `auth`

```solidity
modifier auth(bytes32 _permissionId) 
```

| Input | Type | Description |
|:----- | ---- | ----------- |
| _permissionId | bytes32 | The hash of the permission identifier |

*Auth modifier used in all components of a DAO to check the permissions.*

#### public function `supportsInterface`

> adds a IERC165 to check whether contract supports PluginClones interface or not.

Reviewer: ```suggestion
Checks whether the specified contract supports the PluginClones interface (or the IERC165 interface) or not
```

```solidity
function supportsInterface(bytes4 interfaceId) public view virtual returns (bool) 
```

| Output | Type | Description |
| ------ | ---- | ----------- |

> | [0] | bool | bool whether it supports the IERC165 or PluginClones |

Reviewer: ```suggestion
| [0] | bool | Boolean answer to whether the contract specified supports the IERC165 or the PluginClones interface |
```
